syntax = "proto3";

service QuackMessageAuth {
  rpc Login (LoginMessage) returns (LoginResult) {};
  rpc CreateUser (CreateUserMessage) returns (CreateUserResult) {};
  rpc VerifyEmail (VerifyEmailMessage) returns (VerificationEmailSent) {};
  rpc CheckCode (VerificationCodeMessage) returns (VerificationCodeMatches) {};
}


// For Login
// Argon2 hashes password on server so hash doesn't become the password
message LoginMessage {
  string username = 1;
  string password = 2;
}

message LoginResult {
  bool success = 1;
  string auth_token = 2;
}


// Same as login, can add more info

message VerifyEmailMessage {
  string email = 1;
}

message VerificationEmailSent {
  bool emailSent = 1;
}

message VerificationCodeMessage {
  int32 code = 1;
}

message VerificationCodeMatches {
  bool verified = 1;
}

message CreateUserMessage {
  string username = 1;
  string password = 2;
}

message CreateUserResult {
  bool success = 1;
  string auth_token = 2;
}
